wire-ios-sync-engine/Source/SessionManager/SessionManager.swift

```diff
@@ -809,17 +811,18 @@ public final class SessionManager: NSObject, SessionManagerType {
             return
         }
 
-        delegate?.sessionManagerWillLogout(error: error, userSessionCanBeTornDown: { [weak self] in
-            let group = self?.dispatchGroup
-            group?.enter()
+        requireInternal(activeUserSession.userId == account.userIdentifier, "User session and account are different")
+
+        delegate?.sessionManagerWillLogout(error: error, userSessionCanBeTornDown: { [dispatchGroup] in
+            dispatchGroup.enter()
 
             activeUserSession.close(deleteCookie: deleteCookie) {
                 if deleteAccount {
-                    self?.deleteAccountData(for: account)
```

Comment:
We want to retain `SessionManager` until the log out is complete to ensure we call `deleteAccountData` on it.